Mathlib/Analysis/Calculus/FDeriv/Analytic.lean

```diff
@@ -294,3 +296,71 @@ theorem CPolynomialOn.iterated_deriv (h : CPolynomialOn 𝕜 f s) (n : ℕ) :
   · simpa only [Function.iterate_succ', Function.comp_apply] using IH.deriv
 
 end deriv
+
+namespace FormalMultilinearSeries
+
+variable (p : FormalMultilinearSeries 𝕜 E F)
+
+/-- This series appears in `HasFPowerSeriesOnBall.fderiv` -/
+noncomputable
+def derivSeries : FormalMultilinearSeries 𝕜 E (E →L[𝕜] F) :=
+  (continuousMultilinearCurryFin1 𝕜 E F : (E[×1]→L[𝕜] F) →L[𝕜] E →L[𝕜] F)
+    |>.compFormalMultilinearSeries (p.changeOriginSeries 1)
+
+open Fintype ContinuousLinearMap in
+theorem derivSeries_apply_diag (n : ℕ) (x : E) :
+    derivSeries p n (fun _ ↦ x) x = (n + 1) • p (n + 1) fun _ ↦ x := by
+  simp only [derivSeries, strongUniformity_topology_eq, compFormalMultilinearSeries_apply,
+    changeOriginSeries, compContinuousMultilinearMap_coe, ContinuousLinearEquiv.coe_coe,
+    LinearIsometryEquiv.coe_coe, Function.comp_apply, ContinuousMultilinearMap.sum_apply, map_sum,
+    coe_sum', Finset.sum_apply, continuousMultilinearCurryFin1_apply, Matrix.zero_empty]
+  convert Finset.sum_const _
+  · rw [Fin.snoc_zero, changeOriginSeriesTerm_apply, Finset.piecewise_same, add_comm]
+  · erw [← card, card_subtype, ← Finset.powersetCard_eq_filter, Finset.card_powersetCard, ← card,
+      card_fin, eq_comm, add_comm, Nat.choose_succ_self_right]
+
+end FormalMultilinearSeries
+
+namespace HasFPowerSeriesOnBall
+
+open FormalMultilinearSeries ENNReal Nat
+
+variable {p : FormalMultilinearSeries 𝕜 E F} {f : E → F} {x : E} {r : ℝ≥0∞}
+  (h : HasFPowerSeriesOnBall f p x r) (y : E)
+
+theorem iteratedFDeriv_zero_apply_diag :
+    iteratedFDeriv 𝕜 0 f x (fun _ ↦ y) = p 0 (fun _ ↦ y) := by
+  convert (h.hasSum <| EMetric.mem_ball_self h.r_pos).tsum_eq.symm
+  · rw [iteratedFDeriv_zero_apply, add_zero]
+  · rw [tsum_eq_single 0 <| fun n hn ↦ by haveI := NeZero.mk hn; exact (p n).map_zero]
+    exact congr(p 0 $(Subsingleton.elim _ _))
+
+open ContinuousLinearMap
+
+private theorem factorial_smul' {n : ℕ} : ∀ {F : Type max u v} [NormedAddCommGroup F]
+    [NormedSpace 𝕜 F] [CompleteSpace F] {p : FormalMultilinearSeries 𝕜 E F}
+    {f : E → F}, HasFPowerSeriesOnBall f p x r →
+    n ! • p n (fun _ ↦ y) = iteratedFDeriv 𝕜 n f x (fun _ ↦ y) := by
```

Comment:
I'm a little bit sad with this lemma, as its statement is not the right one: the right statement is that `iteratedFDeriv 𝕜 n f x v` (where `v` doesn't have to be constant) is equal to the sum over all permutations `sigma` of `p n (v (sigma 1), ..., v (sigma n))`. When you specialize to a constant `v`, you get your formula, but it's impossible to recover the general formula from your lemma in general (in particular in positive characteristic). Is there a hope to get the general formula using your approach?

Reply:
@sgouezel Yes that's exactly what I was saying [here on Zulip](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/The.20Taylor.20series.20of.20an.20entire.20function.20converges.20to.20it/near/418229884). I was able to formulate `factorial_smul` immediately after seeing @MichaelStollBayreuth's Zulip question, but it took more time to convince myself that the stronger statement is also true. I have now formulated the stronger statement [here](https://github.com/leanprover-community/mathlib4/compare/symmetrize_changeOrigin_eq_iteratedDeriv?expand=1#diff-afc1a6ce1a23be3624955d95e5804e193249ed762e40bc3f4adf5cecc0cb9822R105-R107) and it shouldn't be too hard to prove by induction, but have not yet found time to tackle the proof, and I realized I should extract an equality between two series, and generalize `symmetrize` to arbitrary DistribMulAction (by a group) on an AddCommMonoid (people may have defined this this when doing group cohomology / representation ...). I agree that the current proof is a "hack" and should be refactor using the stronger statement.

Reply:
So would it be OK to go ahead with this PR and do the refactor later?

Reply:
That's fine with me!